core/common/result.hpp

```diff
@@ -164,6 +164,17 @@ namespace kagome::expected {
                           });
   }
 
+  // Factory methods for avoiding type specification
+  template <typename T>
+  Value<T> makeValue(T &&value) {
+    return Value<T>{std::forward<T>(value)};
+  }
+
+  template <typename E>
+  Error<E> makeError(E &&error) {
+    return Error<E>{std::forward<E>(error)};
+  }
+
```

Comment:
since we use C++17, we agreed to use template deduction guides for this.
you don't need to write `makeValue`, just write `Value(t)`